astropy/modeling/fitting.py

```diff
@@ -1228,8 +1228,14 @@ def _wrap_deriv(params, model, weights, x, y, z=None):
             return [np.ravel(_) for _ in residues]
         else:
             if z is None:
-                return [np.ravel(_) for _ in np.ravel(weights) *
-                        np.array(model.fit_deriv(x, *params))]
+                flat = np.ravel(weights)
```

Comment:
Does this work if the input values are multidimensional? For example, if the test in test_fitters has tx = a 10x15 array, and the same for ty and tw? Even if so, we should have a test of such a case in the test_fitters.py file.

Reply:
I does not work when the input values are multidimensional. I do not believe that it has ever properly worked when the inputs are multi-dimensional.

Reply:
Commit 33908ee may address your issue.